zkevm-circuits/src/rlp_decoder_tables.rs

```diff
@@ -0,0 +1,764 @@
+    ) -> Result<(), Error> {
+        let mut randomness = F::ZERO;
+        challenges.keccak_input().map(|r| randomness = r);
+
+        layouter.assign_region(
+            || "load rlp r_mult table",
+            |mut region| {
+                (0..=MAX_BYTE_COLUMN_NUM).try_for_each(|i| {
+                    region.assign_fixed(
+                        || "table tag",
+                        self.table_tag,
+                        i,
+                        || Value::known(F::from(RlpDecoderFixedTableTag::RMult as u64)),
+                    )?;
+                    region.assign_fixed(
+                        || "pow",
+                        self.length,
+                        i,
+                        || Value::known(F::from(i as u64)),
+                    )?;
+                    region.assign_fixed(
+                        || "r_mult",
+                        self.r_mult,
+                        i,
+                        || Value::known(randomness.pow(&[i as u64, 0, 0, 0])),
+                    )?;
+                    Ok(())
+                })
+            },
+        )?;
+        Ok(())
+    }
+}
+
+/// for value range lookup
+#[derive(Clone, Debug)]
+pub struct RangeTable<const N: usize> {
+    /// table tag
+    pub table_tag: Column<Fixed>,
+    /// value in range
+    pub value: Column<Fixed>,
+}
+
+impl<const N: usize> RangeTable<N> {
+    /// Construct a new RlpDecoderTable
+    pub fn construct<F: Field>(meta: &mut ConstraintSystem<F>) -> Self {
+        Self {
+            table_tag: meta.fixed_column(),
+            value: meta.fixed_column(),
+        }
+    }
+
+    /// build from existed columns
+    pub fn build_from_columns(columns: &[Column<Fixed>]) -> Self {
+        assert!(columns.len() > 2);
+        Self {
+            table_tag: columns[0],
+            value: columns[1],
+        }
+    }
+
+    /// Get the row num of the table
+    pub fn table_size() -> usize {
+        match N {
+            0..=256 => 256,
+            _ => unimplemented!(),
+        }
+    }
+
+    /// Assign the values of the table to the circuit
+    pub fn load<F: Field>(
+        &self,
+        layouter: &mut impl Layouter<F>,
+        challenges: &Challenges<Value<F>>,
+    ) -> Result<(), Error> {
+        let mut randomness = F::ZERO;
+        challenges.keccak_input().map(|r| randomness = r);
+
+        let tag = match N {
+            0..=256 => RlpDecoderFixedTableTag::Range256 as u64,
+            _ => unimplemented!(),
+        };
+        layouter.assign_region(
+            || format!("load rlp range {} table", N),
+            |mut region| {
+                (0..N).try_for_each(|i| {
+                    region.assign_fixed(
+                        || "table tag",
+                        self.table_tag,
+                        i,
+                        || Value::known(F::from(tag)),
+                    )?;
+                    region.assign_fixed(
+                        || "pow",
```

Comment:
I think this is a copy-paste mistake
```suggestion
                        || "range value",
```